lib/bundler/cli.rb

```diff
@@ -69,7 +71,26 @@ def self.deprecated_option(*args, &blk)
     def cli_help
       version
       Bundler.ui.info "\n"
-      self.class.help(shell)
+
+      primary_commands = ["install", "update",
+                          Bundler.feature_flag.cache_command_is_package? ? "cache" : "package",
+                          "exec", "config", "help"]
+
+      list = self.class.printable_commands(true)
+      by_name = list.group_by {|name, _message| name.match(/^bundle (\w+)/)[1] }
+      utilities = by_name.keys.sort - primary_commands
+      primary_commands.map! {|name| (by_name[name] || raise("no primary command #{name}")).first }
+      utilities.map! {|name| by_name[name].first }
+
+      shell.say "Bundler commands:\n\n"
+
+      shell.say "  Primary commands:\n"
+      shell.print_table(primary_commands, :indent => 4, :truncate => true)
+      shell.say
+      shell.say "  Utilities:\n"
+      shell.print_table(utilities, :indent => 4, :truncate => true)
+      shell.say
+      self.class.send(:class_options_help, shell)
```

Comment:
would be nice if this wasnt a `send`

Reply:
it's private, but since we vendor Thor it should be safe

Reply:
That might make thor difficult to upgrade in the future. What does `class_options_help` do and is there an alternative?

Reply:
It formats and prints all of the class options

Reply:
Ah, hrm. You'd think that would be a public thing